api/envoy/extensions/filters/network/mysql_proxy/v3/mysql_proxy.proto

```diff
@@ -15,15 +18,79 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // MySQL Proxy :ref:`configuration overview <config_network_filters_mysql_proxy>`.
 // [#extension: envoy.filters.network.mysql_proxy]
 
+// Auth information of MySQL.
+message AuthInfo {
+  config.core.v3.DataSource username = 1 [(udpa.annotations.sensitive) = true];
+
+  config.core.v3.DataSource password = 2 [(udpa.annotations.sensitive) = true];
+}
+
+// [#next-free-field: 6]
 message MySQLProxy {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.config.filter.network.mysql_proxy.v1alpha1.MySQLProxy";
 
+  message DatabaseRoutes {
+    message Route {
+      // String database name that must exact match with client connection db name. https://dev.mysql.com/doc/refman/5.7/en/identifier-length.html
+      string database = 1 [(validate.rules).string = {max_bytes: 64}];
+
+      // Upstream cluster to forward the command to.
+      string cluster = 2 [(validate.rules).string = {min_len: 1}];
+    }
+
+    // [#not-implemented-hide:] List of database routes.
+    repeated Route routes = 1;
+
+    // Optional catch-all route to forward connection that not specific database name.
+    Route catch_all_route = 2;
+  }
+
   // The human readable prefix to use when emitting :ref:`statistics
   // <config_network_filters_mysql_proxy_stats>`.
   string stat_prefix = 1 [(validate.rules).string = {min_len: 1}];
 
   // [#not-implemented-hide:] The optional path to use for writing MySQL access logs.
   // If the access log field is empty, access logs will not be written.
   string access_log = 2;
+
+  // [#not-implemented-hide:] enable manage MySQL protocol at proxy. Default is false.
+  // * When enable_manage_protocol is true, MySQL filter is a terminal filter, it's config should put at the last of filter chain, in the same time, MySQLProtocolOptions should be set at corresponding config of cluster for connecting to upstream database.
+  // * When enable_manage_protocol is false, MySQL filter is not a terminal filter, it's config can not put at the last of filter chain, the routes and downstream_auth_info fields will be ignore.
+  bool enable_manage_protocol = 3;
```

Comment:
this should be implied by whether `database_routes` is set or not. I don't think you need the flag itself?